src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -37,16 +37,17 @@ protected function createController($controller)
     {
         if (false === strpos($controller, '::') && 2 === substr_count($controller, ':')) {
             // controller in the a:b:c notation then
-            $controller = $this->parser->parse($controller);
-        }
-
-        $resolvedController = parent::createController($controller);
+            $deprecatedNotation = $controller;
+            $controller = $this->parser->parse($deprecatedNotation);
 
-        if (1 === substr_count($controller, ':') && is_array($resolvedController)) {
-            $resolvedController[0] = $this->configureController($resolvedController[0]);
+            @trigger_error(sprintf(
+                'Referencing controllers with %s is deprecated since version 4.1 and will be removed in 5.0. Use %s instead.',
+                $deprecatedNotation,
+                $controller
+            ), E_USER_DEPRECATED);
         }
 
-        return $resolvedController;
+        return parent::createController($controller);
```

Comment:
The whole class can be deprecated after removing the deprecated code in 5.1. Which means removing this class in 6.0. What about deprecating the class right now. Upgrading could then be done by 1/ removing all old notation usage 2/ switching to the Component class equivalent. WDYT?

Reply:
You mean 4.1 and 5.0 right :)

Reply:
@fabpot we still need to class for the `ContainerAwareInterface` logic, see `configureController`. So we can't deprecate it now. We might want to remove ContainerAware logic in the future as well, but that's a different issue.

Reply:
Indeed, I missed that. It could be deprecated at some point, but definitely not part of this work.

Reply:
Can't we move the ContainerAwareInterface logic to the ContainerControllerResolver of the component ?

Reply:
I don't think it's worth it because there is also logic for the AbstractController which is part of the framework bundle. So it fits in the framework bundle.